File: src/src/org/github/olloginov/ideataskbrowser/TaskBrowserNotifier.java
package org.github.olloginov.ideataskbrowser;

import com.intellij.notification.Notification;
import com.intellij.notification.NotificationDisplayType;
import com.intellij.notification.NotificationType;
import com.intellij.notification.Notifications;
import com.intellij.notification.NotificationsConfiguration;

public interface TaskBrowserNotifier {
    String NOTIFICATION_GROUP = "Task Browser";

    void error(String title, String text);

    void info(String title, String text);
}

class TaskBrowserNotifierImpl implements TaskBrowserNotifier {
    public TaskBrowserNotifierImpl() {
        NotificationsConfiguration.getNotificationsConfiguration().register(NOTIFICATION_GROUP, NotificationDisplayType.BALLOON, true);
    }

    private void notify(String title, String message, NotificationType messageType) {
        Notifications.Bus.notify(new Notification(NOTIFICATION_GROUP, title, message, messageType));
    }

    public void error(String title, String text) {
        notify(title, text, NotificationType.ERROR);
    }

    public void info(String title, String text) {
        notify(title, text, NotificationType.INFORMATION);
    }
}

File: src/main/kotlin/org/github/olloginov/ideataskbrowser/TaskBrowserNotifier.kt
package org.github.olloginov.ideataskbrowser

import com.intellij.notification.Notification
import com.intellij.notification.NotificationDisplayType
import com.intellij.notification.NotificationType
import com.intellij.notification.Notifications
import com.intellij.notification.NotificationsConfiguration

const val NOTIFICATION_GROUP: String = "Task Browser"

interface TaskBrowserNotifier {
	fun error(title: String, text: String)

	fun info(title: String, text: String)
}

class TaskBrowserNotifierImpl : TaskBrowserNotifier {
	init {
		NotificationsConfiguration.getNotificationsConfiguration().register(NOTIFICATION_GROUP, NotificationDisplayType.BALLOON, true)
	}

	private fun notify(title: String, message: String, messageType: NotificationType) {
		Notifications.Bus.notify(Notification(NOTIFICATION_GROUP, title, message, messageType))
	}

	override fun error(title: String, text: String) {
		notify(title, text, NotificationType.ERROR)
	}

	override fun info(title: String, text: String) {
		notify(title, text, NotificationType.INFORMATION)
	}
}
